libecho/include/EchoProtocol.h

```diff
@@ -42,19 +47,32 @@ class EchoProtocol {
 
     /**
      * @brief Closes a connection.
+     * Connection that was once opened should be closed (even when one of read/write
+     * functions throws an exception).
      */
     void close();
 
     /**
-     * Reads at most count bytes from audio input.
-     * Can receive 0 bytes, but if it doesn't receive any signal for timeout seconds,
-     * it throws an exception.
+     * @brief           Reads at most @p count bytes from audio input and fills buffer with them.
+     * @param buf       pointer to buffer that should be filled
+     * @param count     number of bytes to read
+     * @param timeout   number of seconds; if no signal is detected by then, function assumes
+     *                  that connection is broken;
+     *                  timeout < 0 means that function can wait infinitely
+     *
+     * @return          Returns number of read bytes (can be any from 0 to count),
+     *                  or -1 in case of failure or end of stream.
```

Comment:
- Warto powiedzieć dlaczego ta liczba może być mniejsza, co znaczy zwrócone 0, czy przy wielu wywołaniach z rzędy może cały czas zwracać 0.
- "-1 in case of failure or end of stream." - jak rozróżnić która sytuacja nastąpiła?

Reply:
Nie jest napisane w jakim stanie jest protokół po zwróceniu -1 - czy trzeba zrobić close()? Czy można jeszcze raz wywołać read(), czy trzeba zawsze lączyć się na nowo itp